### PR TITLE
[Filebeat][udp] - Remove unnecessary multiplication

### LINKED ISSUE #41131: [Filebeat] udp input - unneccessary multiplication by KiB
https://github.com/elastic/beats/blob/56a17e0eaa8c6c9ca9f00b844f25d9731839bb69/filebeat/inputsource/udp/server.go#L65-L67

There is no need to multiple the `config.ReadBuffer` by the size of KiB. The value is already converted into the user's specified unit type when the config is unmarshaled. And secondly, the resulting value is only used in the context of checking if it is non-zero.

So just remove the entire declaration of the `socketSize` variable in this code.

### PR DESCRIPTION
**Pull request by vinit-chauhan**
## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #41131 

## Screenshots

image

### CODE REVIEW

**Review by rdner — APPROVED** [review]

**Review by kush-elastic — APPROVED** [review]

**Comment by pierrehilbert** [comment]
run docs-build

File: filebeat/inputsource/udp/server.go [hunk 1]
```diff
@@ -62,12 +60,11 @@ func (u *Server) createConn() (net.PacketConn, error) {
 	if err != nil {
 		return nil, err
 	}
-	socketSize := int(u.config.ReadBuffer) * humanize.KiByte
-	if socketSize != 0 {
-		if err := listener.SetReadBuffer(int(u.config.ReadBuffer)); err != nil {
-			return nil, err
-		}
+
+	if err := listener.SetReadBuffer(int(u.config.ReadBuffer)); err != nil {
```
Comment (leehinman) [inline filebeat/inputsource/udp/server.go:64]: Is it truly safe to remove the check for 0?  The config only validates that it is positive:

https://github.com/elastic/go-ucfg/blob/587fe67620f33b205484ef84b547d1ad30531035/validator.go#L249-L276

Comment (rdner) [reply filebeat/inputsource/udp/server.go:64]: Good question! Eventually, it's passed to a OS specific implementation, for example this is for POSIX https://cs.opensource.google/go/go/+/master:src/net/sockopt_posix.go;l=75-79?q=setReadBuffer&ss=go%2Fgo

We cannot guarantee that every OS would return us a proper error when the buffer size is 0. I think we should put the zero check back here.

**Review by rdner — CHANGES_REQUESTED** [review]

File: filebeat/inputsource/udp/server.go (hunk 1 above)
Comment (vinit-chauhan) [reply filebeat/inputsource/udp/server.go:64]: Makes sense!! Let me add the condition back.

**Review by leehinman — APPROVED** [review]
LGTM.  Probably needs a changelog entry in bugfix section.

**Comment by vinit-chauhan** [comment]
Hey @leehinman - We are just removing a variable declaration that's it. We are not adding or modifying any features. So should I still add an entry? 

If yes, that would go under dev changelog, correct?

**Review by rdner — APPROVED** [review]
I think this change is alright without a changelog entry, since there is no change of behavior.

**Comment by leehinman** [comment]
> Hey @leehinman - We are just removing a variable declaration that's it. We are not adding or modifying any features. So should I still add an entry?

No changelog needed then.

**Comment by pierrehilbert** [comment]
run docs-build

**Pull request merged**